src/Data/Parameterized/TH/GADT.hs

```diff
@@ -532,3 +681,40 @@ matchShowCtor p con = showCon p (constructorName con) (length (constructorFields
 --
 -- The use of 'DataArg' says that the type parameter of the 'NatRepr' must
 -- be the same as the second type parameter of @T@.
+
+-- $reprs
+--
+-- When working with data kinds with run-time representatives, we encourage
+-- users of @parameterized-utils@ to use the following convention. Given a data
+-- kind defined by
+--
+-- @
+-- data T = ...
+-- @
+--
+-- users should also supply a GADT @TRepr@ parameterized by @T@, e.g.
+--
+-- @
+-- data TRepr (t :: T) where ...
+-- @
+--
+-- Each constructor of @TRepr@ should correspond to a constructor of @T@. If @T@
+-- is defined by
+--
+-- @
+-- data T = A | B Nat
+-- @
+--
+-- we have a corresponding
+--
+-- @
+-- data TRepr (t :: T) where
+--   ARepr :: TRepr 'A
+--   BRepr :: NatRepr w -> TRepr ('B w)
+-- @
+--
+-- Assuming the user of @parameterized-utils@ follows this convention, we
+-- provide the template haskell constructs 'mkRepr' to automate the creation of
+-- the @TRepr@ GADT type. We also provide 'mkKnownReprs', which generates
+-- 'KnownRepr' instances for that GADT type. See the documentation for those two
```

Comment:
"GADT type" is a bit like "ATM machine" in that the last word is redundant :)

Reply:
Good point :)